src/pages/RepoPage/FailedTestsTab/FailedTestsTab.tsx

```diff
@@ -109,7 +106,63 @@ export default function FailedTestsTab() {
         </p>
       </div>
       <SetupOptionSelector />
-      <Content />
+      {usesCli ? <CodecovCLI /> : <GitHubActions />}
     </div>
   )
 }
+
+const Loader = () => (
+  <div className="flex justify-center py-8">
+    <Spinner />
+  </div>
+)
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+function FailedTestsTab() {
+  const { provider, owner, repo } = useParams<URLParams>()
+  const { data: repoOverview } = useRepoOverview({ provider, owner, repo })
+
+  if (repoOverview?.testAnalyticsEnabled) {
+    return (
+      <SentryRoute
+        path={[
+          '/:provider/:owner/:repo/tests',
+          '/:provider/:owner/:repo/tests/:branch',
+        ]}
+        exact
+      >
+        <Suspense fallback={<Loader />}>
+          <div className="flex flex-1 flex-col gap-4">
+            <FailedTestsTable />
+          </div>
+        </Suspense>
+      </SentryRoute>
+    )
+  }
+
+  return (
+    <Switch>
+      <SentryRoute path="/:provider/:owner/:repo/tests/new" exact>
+        <OnboardingWrapper usesCli={false} />
```

Comment:
nit: Can you rename this to `useCLI`

Reply:
Hrrm, I have slight hesitation about this, typically we `use` is reserved for hooks only when writing React apps, you can read more about it here [rules of hooks](https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level). This may end up being a little confusing if people don't have context and are only quickly looking over it.